FormalisingMathematics2025/Section03functions/Sheet1.lean

```diff
@@ -48,44 +43,34 @@ example : ℕ := 2
 -- # But this isn't maths!
 
 -- The type `Prop` contains `Prop`ositions...
-
 #check 2 + 2 = 4
 #check rexp 1 < π
-
 #check 2 + 2 = 5
 #check Irrational (rexp 1 + π)
 #check myFavouriteNumber = yourFavouriteNumber
-
 def MyDifficultProposition : Prop := ∀ n : ℕ, ∃ p, n ≤ p ∧ Prime p ∧ Prime (p + 2)
 def MyEasyProposition : Prop := ∀ n : ℕ, ∃ p, n ≤ p ∧ Prime p ∧ Prime (p + 2) ∧ Prime (p + 4)
-def MyVeryEasyProposition : Prop := ∀ n : ℕ, ∃ p, n ≤ p
 
 -- Key! If `p : Prop`, an expression of type `p` is a proof of `p`.
-
 example : 2 + 2 = 4 := rfl
 example : 2 + 2 ≠ 5 := by simp
 example : ∀ n : ℕ, 2 ≤ n → ∃ x y z : ℕ, 4 * x * y * z = n * (x * y + x * z + y * z) := sorry
--- Erdős-Strauss conjecture
-
-example (n : ℕ) (hn : 2 ≤ n) :
-  ∃ x y z : ℕ, 4 * x * y * z = n * (x * y + x * z + y * z) := sorry
```

Comment:
I don't understand what this example tries to convey that isn't already conveyed earlier. Maybe accepting parameters? But that isn't a specific feature of proofs per se. Maybe could be explained earlier separately? Anyway I felt this is distracting from the most important point below so I dropped it